clients/src/test/java/org/apache/kafka/clients/MockClient.java

```diff
@@ -65,8 +66,8 @@ public FutureResponse(Struct responseBody, boolean disconnected, RequestMatcher
     private Node node = null;
     private final Set<String> ready = new HashSet<>();
     private final Map<Node, Long> blackedOut = new HashMap<>();
-    private final Queue<ClientRequest> requests = new ArrayDeque<>();
-    private final Queue<ClientResponse> responses = new ArrayDeque<>();
+    private final Queue<ClientRequest> requests = new ConcurrentLinkedDeque<>();
```

Comment:
It was needed for the added test because the main thread adds a response the consumer thread then reads. I'll remove the consumer thread and revert this instead.